File: ServiceMaxiPad/Classes/ProductIQ/www/console.js
(function(){
	//var c = window.console; // backup
 
	function log(type, message){
		BRIDGE.invoke("Console", "log", {type : type, message : message});
	}
 
 	console.info = function(message) { log("INFO", message); };
	console.log = function(message) { log("LOG", message); };
	console.error = function(message) { log("ERROR", message); };
	console.warn = function(message) { log("WARN", message); };
	console.debug = function(message) { log("DEBUG", message); };
 
})();

File: ServiceMaxiPad/Classes/ProductIQ/POC/Console.m
//
//  Console.m
//  ProductIQ
//
//  Created by Indresh M S on 4/19/15.
//  Copyright (c) 2015 ServiceMax. All rights reserved.
//

#import "Console.h"

@implementation Console

- (void)log:(NSString *) params {
    [self logMessage:[self parse:params]];
}

- (void)error:(NSString *) params {
    [self logMessage:[self parse:params]];
}

- (void)warn:(NSString *) params {
    [self logMessage:[self parse:params]];
}

- (void)debug:(NSString *) params {
    [self logMessage:[self parse:params]];
}

- (void)info:(NSString *) params {
    [self logMessage:[self parse:params]];
}

- (NSDictionary *)parse:(NSString *) str {
    NSError *error = nil;
    NSDictionary *ret =
    [NSJSONSerialization JSONObjectWithData:[str dataUsingEncoding:NSUTF8StringEncoding] options:0 error:&error];
    return ret;
}

-(void)logMessage:(NSDictionary *) details {
    NSLog(@"%@::%@", details[@"type"], details[@"message"]);
}


@end
